### PR TITLE
Fix the `checks-integration` errors on LightGBMTuner

### PR DESCRIPTION
**Pull request by c-bata**
## Motivation

Fix mypy errors detected at:
https://github.com/optuna/optuna/actions/runs/6165757759/job/16734112261

## Description of the changes

Fix following errors:

```
Run mypy . --warn-unused-ignores
optuna/integration/lightgbm.py:93: error: Item "None" of "Optional[List[Tuple[str, str, float, bool]]]" has no attribute "__iter__" (not iterable)  [union-attr]
optuna/integration/lightgbm.py:101: error: Incompatible return value type (got "Union[Tuple[str, str, float, bool], Any]", expected "Optional[List[Any]]")  [return-value]
optuna/integration/lightgbm.py:[11](https://github.com/optuna/optuna/actions/runs/6165757759/job/16734112261#step:11:12)1: error: Argument 1 to "len" has incompatible type "Optional[List[Tuple[str, str, float, bool]]]"; expected "Sized"  [arg-type]
optuna/integration/lightgbm.py:111: error: Value of type "Optional[List[Tuple[str, str, float, bool]]]" is not indexable  [index]
Found 4 errors in 1 file (checked 335 source files)
Error: Process completed with exit code 1.
```

### CODE REVIEW

File: optuna/integration/lightgbm.py [hunk 1]
```diff
@@ -88,27 +93,35 @@ def __init__(
         self._report_interval = report_interval
 
     def _find_evaluation_result(
-        self, target_valid_name: str, env: "CallbackEnv"
-    ) -> Optional[List]:
-        for evaluation_result in env.evaluation_result_list:
+        self, target_valid_name: str, env: CallbackEnv
+    ) -> _LGBM_BoosterEvalMethodResultType | None:
```
Comment (c-bata) [inline optuna/integration/lightgbm.py:97]: See
https://github.com/microsoft/LightGBM/blob/921479b99fb5b691801e0e794f2196a94ea17d79/python-package/lightgbm/callback.py#L57

**Comment by c-bata** [comment]
@Alnusjaponica @eukaryo Could you review this PR?

**Comment by eukaryo** [comment]
LGBM.

**Comment by Alnusjaponica** [comment]
I confirmed this PR fixes mypy error. LGTM.
@eukaryo Could you merge this PR?

**Review by eukaryo — APPROVED** [review]

**Pull request merged**